mono/mini/basic.cs

```diff
@@ -1428,4 +1428,12 @@ public static int test_1_rem_un_cfold ()
 
 		return k == -32768 ? 0 : 1;
 	}
+
+	static int variable_with_constant_address;
+
+	public static int test_cfold_with_non_constant_ternary_op () {
+		variable_with_constant_address = 0;
+		var old = System.Threading.Interlocked.CompareExchange(ref variable_with_constant_address, 1, 0);
+		return old == 0 && variable_with_constant_address == 1 ? 0 : 1;
+	}
```

Comment:
This should go to objects.cs since it uses statics/calls. Also, it should be named test_<EXPECTED RETURN VALUE>_...